web/react/components/textbox.jsx

```diff
@@ -279,7 +292,16 @@ export default class Textbox extends React.Component {
         this.doProcessMentions = true;
     }
 
+    showPreview(e) {
+        e.preventDefault();
+        e.target.blur();
+        this.setState({preview: !this.state.preview});
+        this.resize();
+    }
+
     render() {
+        let previewLinkVisible = (this.value || '').length > 0;
```

Comment:
Related to above comment, why not just directly access `this.props.messageText` here?